File: src/cmd/co/co.c
#include "../../stacks/C.h"
#include "../../main/C.h"
#include "../../misc/C.h"

#define bcopy(s, d, n) memcpy(d, s, n)

/* Subroutine */
void co_()
{
    /* Initialized data */

    static int cmdsav = -1;

    int init, nDragon=0;

#include "co.dc.h"
#include "co.ec.h"
    bcopy(use_.muse, use_.nuse, sizeof(use_.nuse));
    if (*locad != 1 || *locsm != 2)  {
        gstak_((int *)locad, (int *)locsm);
    }
    if (cmnd_.cmdsig != cmdsav) {
	cmdsav = cmnd_.cmdsig;
	init = 1;
    } else {
	init = 0;
    }
    if (*addf != 0) {
	init += 8;
    }
    if (*acf != 0) {
	init += 32;
    }
    if (*dcf != 0) {
	init += 4;
    }
    cmbn_(&init);
    stk_[0] = stk_[1];
    if (use_.kusfl != 0) {
	bcopy(stk_[1].kuse, use_.nuse, sizeof(stk_[0].kuse));
    }
    main_.pltd = 0;
} /* co_ */

File: src/cmd/co/co.f
	subroutine co

	include '../../stacks/F.h'
	include '../../main/F.h'
	include 'co.df.h'
c
	integer init
	automatic init
	integer cmdsav
	data cmdsav/-1/
c
	include 'co.if.h'
c
	include 'co.ef.h'
c
	call move(muse,nuse,maxchn/8)
c  call gstack(locad,locsm)
	if(locad.ne.1.or.locsm.ne.2)call gstak(locad,locsm)
	if(cmdsig.ne.cmdsav) then
		cmdsav=cmdsig
		init=1
	else
		init=0
	endif
	if(addf.ne.0) init=init+8
	if(acf.ne.0) init=init + 32
	if(dcf.ne.0) init=init + 4
	call cmbn(init)
	call move(numpt2,numpt1,stklen)
	if(kusfl.ne.0) call move(kuse2,nuse,stkuse/8)
	pltd=0
	end
